Doc/library/stdtypes.rst

```diff
@@ -4844,19 +4848,22 @@ mistakes like ``dict[str][str]``::
      File "<stdin>", line 1, in <module>
    TypeError: There are no type variables left in dict[str]
 
-However, that syntax is valid when type variables are used::
+However, that expression is valid when :ref:`type variables <generics>` are
+used.  ``GenericAlias`` can only be indexed if it has type variables.  The index
+must have as many elements as there are type variable items in the
+``GenericAlias`` object's :attr:`__args__ <genericalias.__args__>`. ::
 
    >>> from typing import TypeVar
-   >>> T = TypeVar('T')
-   >>> dict[str, T][int]
```

Comment:
Maybe keep this example too? It is important because it shows you can have some type variables and some concrete types in the parameters.